deepseg.cc

```diff
@@ -225,11 +202,23 @@ void init_tensorflow(calcinfo_t &info) {
 
 	// create Mat for small mask
 	info.ofinal = cv::Mat(info.output.rows,info.output.cols,CV_8UC1);
+
+	info.lock_raw = PTHREAD_MUTEX_INITIALIZER;
+	info.lock_mask = PTHREAD_MUTEX_INITIALIZER;
+	info.start = false;
 }
 
-void calc_mask(calcinfo_t &info, timinginfo_t &ti) {
+void calc_mask(calcinfo_t &info) {
+	timestamp_t t0 = timestamp();
+
+	// copy frame from shared buffer
+	cv::Mat raw;
+	pthread_mutex_lock(&info.lock_raw);
+	raw = info.raw.clone();
```

Comment:
Can we avoid the double cloning (in/out of shared buffer)? Unsure of performance impact...

Reply:
Structurally we could pre-allocate a few picture buffers and just hand out the pointers to them. This would require proper tracking which pointers are in use at any time and would require some preparations not yet present in the current code.